### DISCUSSION TITLE
Shortuct for traing and testing loop

### DISCUSSION THREAD

**Discussion by zillur01** [mrdbourke/pytorch-deep-learning#299] [Q&A] [answered]
Video: PyTorch workflow

I see for training and testing, a loop is required and loss function, optimizer, and everything has to be defined explicitly. I used `TensorFlow` before and it does all these steps in only 1-2 lines of code like `model.fit()`. Is there any such function in `PyTorch`?

**Comment by mrdbourke** [comment, marked as answer]
Hi @zillur-av,

Good question!

Yes there is a few options for this:
1. Create your own functions for PyTorch training/testing loops, as seen here: https://www.learnpytorch.io/05_pytorch_going_modular/
2. Use a framework such as [Lightning AI](https://github.com/Lightning-AI/lightning) or [FastAI](https://github.com/fastai/fastai) or [MosaicML](https://www.mosaicml.com/) to do the training loop for you

Otherwise, there's no native `model.fit()` inside PyTorch (I'm pretty sure this is by design, to add as much hackability to PyTorch as possible).
